presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -201,6 +204,42 @@ protected Node visitShowTables(ShowTables showTables, Void context)
         }
 
         @Override
+        protected Node visitShowGrant(ShowGrant showGrant, Void context)
+        {
+            String catalogName = session.getCatalog().orElse(null);
+            Expression predicate = equal(nameReference("grantee"), new StringLiteral(session.getUser()));
+
+            Optional<String> identity = showGrant.getIdentity();
+            if (identity.isPresent() && !identity.get().equals(session.getUser())) {
+                denyShowGrant(identity.get());
+            }
+
+            Optional<QualifiedName> tableName = showGrant.getTableName();
+            if (tableName.isPresent()) {
+                QualifiedObjectName qualifiedTableName = createQualifiedObjectName(session, showGrant, tableName.get());
+
+                Expression tablePredicate = equal(nameReference("table_name"), new StringLiteral(qualifiedTableName.getObjectName()));
+                predicate = logicalAnd(predicate, tablePredicate);
+            }
+
+            if (catalogName == null) {
```

Comment:
I don't know of such a way, but will check again. Will keep this in mind when documenting.